src/middlewared/middlewared/etc_files/local/avahi/avahi-daemon.conf

```diff
@@ -0,0 +1,52 @@
+# avahi is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# avahi is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+# or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with avahi; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+# USA.
+# See avahi-daemon.conf(5).
+<%
+    import os
+    import sys
+    logger = middleware.logger
+
+    def ipv4_enabled():
```

Comment:
Just an idea, you could perhaps replace all this method with
any(call_sync('interface.ip_in_use', {'ipv4': true, 'ipv6': false}))